### PR TITLE
LPS-69016 Adds unit test

### PR DESCRIPTION
**Pull request by fernandosouza**

### CODE REVIEW

File: modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-select/src/test/testJS/select_field.js [hunk 1]
```diff
@@ -140,5 +146,190 @@ describe(
...
+
+						inputElement.val('foo');
+
+						triggerEvent('input', inputElement);
+
+						window.setTimeout(
+							function() {
+								var term = container.one('.drop-chosen ul > li b').getContent();
+
+								assert.equal(
```
Comment (brunobasto) [inline modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-select/src/test/testJS/select_field.js]: these can go on a single line.

File: modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-select/src/test/testJS/select_field.js [hunk 2]
```diff
@@ -140,5 +146,190 @@ describe(
...
+
+						inputElement.val('        ');
+
+						triggerEvent('input', inputElement);
+
+						window.setTimeout(
+							function() {
+								var afterSearchList = container.one('.drop-chosen ul').getHTML();
+
+								assert.equal(
```
Comment (brunobasto) [inline modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-select/src/test/testJS/select_field.js]: these can go on a single line

File: modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-select/src/test/testJS/select_field.js [hunk 3]
```diff
@@ -140,5 +146,190 @@ describe(
...
+					function(done) {
+						selectField.set('options', [{label: 'foo', value: 'foo'}, {label: 'bar', value: 'bar'}]);
+
+						var container = selectField.get('container');
+
+						var beforeSearchList = container.one('.drop-chosen ul').getHTML();
+
+						var inputElement = container.one('input');
+
+						inputElement.val('        ');
```
Comment (brunobasto) [inline modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-select/src/test/testJS/select_field.js]: Not clear why all this spacing.

File: modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-select/src/test/testJS/select_field.js [hunk 4]
```diff
@@ -140,5 +146,190 @@ describe(
...
+
+						triggerEvent('input', inputElement);
+
+						selectField.openList();
+
+						window.setTimeout(
+							function() {
+								var items = container.all('.drop-chosen ul > li');
+
+								assert.equal(
```
Comment (brunobasto) [inline modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-select/src/test/testJS/select_field.js]: These can go on a single line.

File: modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-select/src/test/testJS/select_field.js [hunk 5]
```diff
@@ -140,5 +146,190 @@ describe(
...
+
+						selectField.openList();
+
+						window.setTimeout(
+							function() {
+								selectField.closeList();
+
+								var afterSearchList = container.one('.drop-chosen ul').getHTML();
+
+								assert.equal(
```
Comment (brunobasto) [inline modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-select/src/test/testJS/select_field.js]: These can go on a single line.

File: modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-select/src/test/testJS/select_field.js [hunk 6]
```diff
@@ -140,5 +146,190 @@ describe(
...
+						selectField.openList();
+
+						assert.equal(
+							'foo',
+							container.one('input').val()
+						);
+
+						selectField.closeList();
+
+						assert.equal(
```
Comment (brunobasto) [inline modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-select/src/test/testJS/select_field.js]: These can go on a single line.

File: modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-select/src/main/resources/META-INF/resources/select_field.js [hunk 7]
```diff
@@ -90,6 +89,7 @@ AUI.add(
 							{
 								options: instance.get('options'),
 								selecteCaretDoubleIcon: Liferay.Util.getLexiconIconTpl('caret-double-l', 'icon-monospaced'),
```
Comment (brunobasto) [inline modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-select/src/main/resources/META-INF/resources/select_field.js]: **selecte** is also misspelled here

File: modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-select/src/test/testJS/select_field.js (hunk 4 above)
Comment (brunobasto) [inline modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-select/src/test/testJS/select_field.js]: These can go on a single line.

File: modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-select/src/test/testJS/select_field.js [hunk 8]
```diff
@@ -140,5 +146,190 @@ describe(
...
+					function() {
+						selectField.set('options', [{label: 'foo', value: 'foo'}, {label: 'bar', value: 'bar'}]);
+
+						var container = selectField.get('container');
+
+						container.one('input').val('foo');
+
+						selectField.openList();
+
+						assert.equal(
```
Comment (brunobasto) [inline modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-select/src/test/testJS/select_field.js]: These can go on a single line.

File: modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-select/src/test/testJS/select_field.js (hunk 3 above)
Comment (fernandosouza) [inline modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-select/src/test/testJS/select_field.js]: It was only a test but I committed by mistake

**Pull request closed without merging**